### PR TITLE
Upgrade to latest Herman.

### PR DESCRIPTION
**Pull request by jgerigmeyer**

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
[Bug: Herman color preview styles are not loaded in color preview iframes.](https://trello.com/c/jvd58ABb/108-bug-herman-color-preview-styles-are-not-loaded-in-color-preview-iframes)

**Comment by dvdherron** [comment]
@jgerigmeyer Not sure if we want to fix it in another branch or not, but I think I have a fix for removing the background-images from the iframe previews. 

In `/layout/root.scss` if we change `html` to `[data-layout='root']` and add `[data-layout='root']` to the `html` tag in `content/layout/base.njk` the html tags in the iframes won't have the background-images applied.

**Comment by jgerigmeyer** [comment]
@dvdherron This works now, but something in our OddSite `styleguide.scss` is overriding the Herman styles and making the font size really big for color samples. Is this just something we need to adjust in OddSite, or something that Herman should account for?

Screen Shot 2022-11-11 at 12 57 37 PM

**Comment by jgerigmeyer** [comment]
Or @mirisuzanne, if you have a sense for whether the above is something that will or won't need to be addressed in Herman, that'd be helpful. If it's not, then I'd like to go ahead and do a full Herman release now.

**Comment by mirisuzanne** [comment]
@jgerigmeyer the issue is that:

- in some iframes (like examples) we need all that detail from the host project
- in other iframes (like color swatches) we really want as little as possible

I suppose, rather than making herman users set up different imports, we could override what's important. So make sure the swatch fonts are using herman-defined size variables, rather than the default font size in the iframe?

**Comment by jgerigmeyer** [comment]
> I suppose, rather than making herman users set up different imports, we could override what's important. So make sure the swatch fonts are using herman-defined size variables, rather than the default font size in the iframe?

Yeah, that makes sense to me. For colors, I think really we only need the color vars from the source project? But I agree it seems like overriding what we care about is easier than requiring users to have yet another stylesheet.

I'll file a new bug in Herman, and wait before doing a release.

**Comment by jgerigmeyer** [comment]
[Bug: Herman color preview styles are easily overridden by site custom styles.](https://trello.com/c/e6ymMLy8/110-bug-herman-color-preview-styles-are-easily-overridden-by-site-custom-styles)

**Comment by dvdherron** [comment]
> Or @mirisuzanne, if you have a sense for whether the above is something that will or won't need to be addressed in Herman, that'd be helpful. If it's not, then I'd like to go ahead and do a full Herman release now.


> > I suppose, rather than making herman users set up different imports, we could override what's important. So make sure the swatch fonts are using herman-defined size variables, rather than the default font size in the iframe?

I think the herman `--code` variable is part of the issue here. The custom property `--code` for size is conflicting with the OddSite variable (--code). And there's a Herman color named `code` as well. That custom property is new to the Herman color previews. 

> Yeah, that makes sense to me. For colors, I think really we only need the color vars from the source project? But I agree it seems like overriding what we care about is easier than requiring users to have yet another stylesheet.
> 
> I'll file a new bug in Herman, and wait before doing a release.

Currently to show custom properties in the example docs we are generating Herman color variables in `[data-herman-preview='colors']`. The `code` _color_ custom property in Herman is conflicting with the `code` _size_ custom property on OddSite. 

bild

In Herman, instead of generating custom properties for all Herman-specific colors, we should probably only generate custom properties for the few colors we are using in examples in the documentation? (If we removed them all, the custom property example wouldn't work.) And those should be prefixed in a way that wouldn't conflict with someone else's project? 

Otherwise, I think the font also looks bigger because it's a different font. I can play around with the sizing of the text too.

**Comment by stacyk** [comment]
> Otherwise, I think the font also looks bigger because it's a different font. I can play around with the sizing of the text too.

@dvdherron (maybe in addition to some other things you pointed out like the font-size for code is now a color, I also think the size of everything is bigger (see swatch size too) because the html's font-size from OddSite is scaling everything now:

html { ... font-size: calc(1.25em + .5vw); }

has-html-font-size
removed-html-font-size

**Comment by jgerigmeyer** [comment]
@stacyk @dvdherron This looks good to me now -- what do you think?

https://deploy-preview-276--oddleventy.netlify.app/styleguide/config-color

**Comment by jgerigmeyer** [comment]
@mirisuzanne I guess the auto-custom-props logic results in a lot of (unused, in this case) custom properties being added to the iframes -- but I think that's okay?

```
<style>
      [data-herman-preview="colors"] {
  --small:.8rem;
  --xsmall:.7rem;
  --xxsmall:.6rem;
  --rem:calc(1.25em + .5vw);
  --medium:1.25rem;
  --large:calc(1.25rem + .5vw);
  --xlarge:calc(1.75rem + 1vw);
  --h1:var(--xlarge, calc(1.75rem + 1vw));
  --h2:var(--large, calc(1.25rem + 0.5vw));
  --h3:var(--medium, 1.25rem);
  --h4:1.1rem;
  --code:.8em;
  --gutter:.75rem;
  --newline--default:var(--gutter, 0.75rem);
  --newline:var(--newline--default, 0.75rem);
  --gutter-plus:calc(var(--gutter)*1.5);
  --double-gutter:calc(var(--gutter)*2);
  --triple-gutter:calc(var(--gutter)*3);
  --shim:calc(var(--gutter)/2);
  --shim-plus:calc(var(--gutter)*.75);
  --half-shim:calc(var(--gutter)/4);
  --quarter-shim:calc(var(--gutter)/8);
  --spacer:calc(var(--double-gutter) + 3vw);
  --page-margin:calc(var(--quarter-shim) + 4vw);
  --border-width:medium;
  --radius:var(--half-shim);
  --grid-hero:calc(var(--gutter)*8);
  --logo-item:10rem;
  --grid-item:16rem;
  --min-page:22rem;
  --page:30rem;
  --wide-page:38rem;
  --wide-break:56em;
  --extra: 4rem;
  --main-min: calc(100vw - var(--page-margin) * 2);
  --page-columns: [full-start] var(--page-margin) [content-start wide-start page-start] minmax(var(--main-min), var(--page)) [page-end wide-end content-end] var(--page-margin) [full-end];
  --main-min: var(--min-page);
  --page-columns: [full-start] var(--page-margin) [content-start] minmax(0, 1fr) [wide-start] minmax(0, var(--extra)) [page-start] minmax(var(--main-min), var(--page)) [page-end] minmax(0, var(--extra)) [wide-end] minmax(0, 1fr) [content-end] var(--page-margin) [full-end];
  --clip-overlap: var(--page-margin);
  --clip-angle: var(--double-gutter);
  --clip-space: calc(var(--clip-angle) + var(--gutter));
  --clip-left: polygon(var(--clip-angle) 0, 100% 0, 100% 100%, 0 100%);
  --clip-right: polygon( 0 0, 100% 0, calc(100% - var(--clip-angle)) 100%, 0 100% );
  --alert-fade: calc( var(--ccs-mode--zero) * 0.9 + var(--ccs-invert--zero) * 0.8 );
  --aa-search-input-height:44px;
  --aa-input-icon-size:20px;
  --aa-base-unit:16;
  --aa-spacing-factor:1;
  --aa-spacing:calc(var(--aa-base-unit)*var(--aa-spacing-factor)*1px);
  --aa-spacing-half:calc(var(--aa-spacing)/2);
  --aa-panel-max-height:650px;
  --aa-base-z-index:9999;
  --aa-font-size:calc(var(--aa-base-unit)*1px);
  --aa-font-family:inherit;
  --aa-font-weight-medium:500;
  --aa-font-weight-semibold:600;
  --aa-font-weight-bold:700;
  --aa-icon-size:20px;
  --aa-icon-stroke-width:1.6;
  --aa-icon-color-rgb:119,119,163;
  --aa-icon-color-alpha:1;
  --aa-action-icon-size:20px;
  --aa-text-color-rgb:38,38,39;
  --aa-text-color-alpha:1;
  --aa-primary-color-rgb:62,52,211;
  --aa-primary-color-alpha:0.2;
  --aa-muted-color-rgb:128,126,163;
  --aa-muted-color-alpha:0.6;
  --aa-panel-border-color-rgb:128,126,163;
  --aa-panel-border-color-alpha:0.3;
  --aa-input-border-color-rgb:128,126,163;
  --aa-input-border-color-alpha:0.8;
  --aa-background-color-rgb:255,255,255;
  --aa-background-color-alpha:1;
  --aa-input-background-color-rgb:255,255,255;
  --aa-input-background-color-alpha:1;
  --aa-selected-color-rgb:179,173,214;
  --aa-selected-color-alpha:0.205;
  --aa-description-highlight-background-color-rgb:245,223,77;
  --aa-description-highlight-background-color-alpha:0.5;
  --aa-detached-media-query:(max-width:680px);
  --aa-detached-modal-media-query:(min-width:680px);
  --aa-detached-modal-max-width:680px;
  --aa-detached-modal-max-height:500px;
  --aa-overlay-color-rgb:115,114,129;
  --aa-overlay-color-alpha:0.4;
  --aa-panel-shadow:0 0 0 1px rgba(35,38,59,0.1),0 6px 16px -4px rgba(35,38,59,0.15);
  --aa-scrollbar-width:13px;
  --aa-scrollbar-track-background-color-rgb:234,234,234;
  --aa-scrollbar-track-background-color-alpha:1;
  --aa-scrollbar-thumb-background-color-rgb:var(--aa-background-color-rgb);
  --aa-scrollbar-thumb-background-color-alpha:1;
  --aa-spacing-factor:1.2;
  --aa-action-icon-size:22px;
  --aa-text-color-rgb:183,192,199;
  --aa-primary-color-rgb:146,138,255;
  --aa-muted-color-rgb:146,138,255;
  --aa-input-background-color-rgb:0,3,9;
  --aa-background-color-rgb:21,24,42;
  --aa-selected-color-rgb:146,138,255;
  --aa-selected-color-alpha:0.25;
  --aa-description-highlight-background-color-rgb:0 255 255;
  --aa-description-highlight-background-color-alpha:0.25;
  --aa-icon-color-rgb:119,119,163;
  --aa-panel-shadow:inset 1px 1px 0 0 #2c2e40,0 3px 8px 0 #000309;
  --aa-scrollbar-track-background-color-rgb:44,46,64;
  --aa-scrollbar-thumb-background-color-rgb:var(--aa-background-color-rgb);
}
    </style>
```

**Comment by mirisuzanne** [comment]
@jgerigmeyer not a huge issue. And we do allow people to bypass the automation. I think it's ok.

**Comment by stacyk** [comment]
Our layout sizes are saying that var(--gutter) is not defined ... looking into this more 
image

**Comment by jgerigmeyer** [comment]
> Our layout sizes are saying that var(--gutter) is not defined ... looking into this more 
> 

Oh, good catch. I wonder if it (and other custom props?) aren't declared on one of the root nodes?  We may want a custom CSS prop stylesheet... 🤔

**Comment by dvdherron** [comment]
> > Our layout sizes are saying that var(--gutter) is not defined ... looking into this more
> 
> Oh, good catch. I wonder if it (and other custom props?) aren't declared on one of the root nodes? We may want a custom CSS prop stylesheet... 🤔

I think what's going on is that the project custom properties are only being loaded on `[data-herman-preview='colors']`. Should they be loaded under the generic `[data-herman-preview]` instead? Editing this in the html fixes the issues. 

image

Sorry for not catching this the first time @jgerigmeyer !

**Comment by jgerigmeyer** [comment]
> I think what's going on is that the project custom properties are only being loaded on `[data-herman-preview='colors']`. Should they be loaded under the generic `[data-herman-preview]` instead? Editing this in the html fixes the issues.

🤦

**Comment by jgerigmeyer** [comment]
> Our layout sizes are saying that var(--gutter) is not defined

@dvdherron @stacyk Fixed in https://github.com/oddbird/oddleventy/pull/276/commits/5ccf63d6d8c553725d4f160a7a1c29d23e89ef11

**Comment by dvdherron** [comment]
> @stacyk @dvdherron This looks good to me now -- what do you think?
> 
> https://deploy-preview-276--oddleventy.netlify.app/styleguide/config-color

This looks good to me. The project custom properties are showing up in the right places. I would need to play with the colors some to test that all of our variables are showing in previews but I don't know if that should hold this up. We might need to do something to add `[data-ccs-colors]` to the group of custom properties? I have to run now but I can look more into that.

**Review by stacyk — APPROVED** [review]
Much better! Thank you!

**Review by stacyk — COMMENTED** [review]
I hate to bring this up, but now our text sizes aren't rendering the way I'd expect them to... 

text-size-before-now

the current size like xxsmall is way smaller than how it appears on the site.

**Comment by jgerigmeyer** [comment]
> I hate to bring this up, but now our text sizes aren't rendering the way I'd expect them to...
> the current size like xxsmall is way smaller than how it appears on the site.

Good catch! @mirisuzanne @dvdherron Now that we've removed custom app styles from the previews, it looks like we may need some additional font settings added?

**Comment by jgerigmeyer** [comment]
> @jgerigmeyer You mentioned there being a way to see these changes from Herman before we merge them here. Can you point me to how to set that up for the next round of edits?

@dvdherron Sure thing -- a few minor changes are required to run OddLeventy against your local version of Herman:

1. In your local OddLeventy, in [`.sassdocrc`](https://github.com/oddbird/oddleventy/blob/herman-upgrade/.sassdocrc#L1), change `theme: herman` to e.g. `theme: ./../sassdoc-theme-herman/` or wherever the repo exists locally for you.
2. In your local Herman, in [`.yarnrc.yml`](https://github.com/oddbird/sassdoc-theme-herman/blob/main/.yarnrc.yml), add `nodeLinker: node-modules` and then run `yarn`.

That should do it, and your local OddLeventy should use your local Herman for subsequent `yarn build` or `yarn serve` or whatever. Let me know if that doesn't work!

File: src/scss/initial/_root.scss [hunk 1]
```diff
@@ -16,6 +16,6 @@ html {
     0 100%
   );
 
-  font-size: config.size('rem');
+  font-size: calc(1.25em + 0.5vw);
```
Comment (jgerigmeyer) [inline src/scss/initial/_root.scss:19]: @mirisuzanne Feel free to adjust this if you wanted it stored in a custom property still.

**Review by jgerigmeyer — COMMENTED** [review]
@dvdherron @stacyk @mirisuzanne Ready for another round of review! ♻️

**Comment by jgerigmeyer** [comment]
[Bug Herman Relative Font Sizes](https://trello.com/c/t5SVCOo7/112-bug-herman-relative-font-sizes)

**Review by dvdherron — APPROVED** [review]
The sizing and fonts all look better to me. 

One thing I'm curious about that will probably be a separate story when we decide to show color custom properties here, how do we get access to `data-ccs-colors` in previews? 

Generating the tokens on the appropriate tag is clear to me but I'm unclear on where we would put the `data-ccs=`root`, and `data-ccs-colors=""` attributes.

File: src/scss/config/_scale.scss [hunk 2]
```diff
@@ -29,7 +29,6 @@ $font-sizes: (
   'small': 0.8rem,
   'xsmall': 0.7rem,
   'xxsmall': 0.6rem,
-  'rem': calc(1.25em + 0.5vw),
```
Comment (jgerigmeyer) [inline src/scss/config/_scale.scss:32]: I guess another option here, if we think it's valuable to display the root size, would be to replace this with something like `'root': 1rem`?

**Review by mirisuzanne — APPROVED** [review]
I moved the root size back into the config, and added it to the docs - without also adding it to the custom properties. I think this works well!

**Comment by mirisuzanne** [comment]
@dvdherron I think your right that it's a separate story. I _think_ we should be able to solve that by generating a special custom-properties documentation file that sets the colors on `html` instead? Since we also maintain CCS, we should be able to figure something out. :)

**Pull request merged**